src/core/skills/characters/yijiang2014/qiangzhi.ts

```diff
@@ -0,0 +1,146 @@
+import { EventPacker, GameEventIdentifiers, ServerEventFinder } from 'core/event/event';
+import { Sanguosha } from 'core/game/engine';
+import { AllStage, CardUseStage, PhaseStageChangeStage, PlayerPhaseStages } from 'core/game/stage_processor';
+import { Player } from 'core/player/player';
+import { PlayerCardsArea } from 'core/player/player_props';
+import { Room } from 'core/room/room';
+import { CommonSkill, ShadowSkill, TriggerSkill } from 'core/skills/skill';
+import { PatchedTranslationObject, TranslationPack } from 'core/translations/translation_json_tool';
+
+@CommonSkill({ name: 'qiangzhi', description: 'qiangzhi_description' })
+export class QiangZhi extends TriggerSkill {
+  isTriggerable(event: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>, stage?: AllStage) {
+    return stage === PhaseStageChangeStage.StageChanged;
+  }
+
+  canUse(room: Room, owner: Player, content: ServerEventFinder<GameEventIdentifiers.PhaseStageChangeEvent>) {
+    return content.playerId === owner.Id && content.toStage === PlayerPhaseStages.PlayCardStageStart;
+  }
+
+  async onTrigger() {
+    return true;
+  }
+
+  async onEffect(room: Room, content: ServerEventFinder<GameEventIdentifiers.SkillEffectEvent>) {
+    const ownerId = content.fromId;
+
+    room.notify(
+      GameEventIdentifiers.AskForChoosingPlayerEvent,
+      EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingPlayerEvent>({
+        toId: ownerId,
+        players: room.AlivePlayers.filter(
+          player => player.getCardIds(PlayerCardsArea.HandArea).length > 0 && player.Id !== ownerId,
+        ).map(player => player.Id),
+        requiredAmount: 1,
+        conversation: TranslationPack.translationJsonPatcher(
+          '{0}: please choose a player to show a hand card',
+          this.GeneralName,
+        ).extract(),
+      }),
```

Comment:
add `triggeredBySkills`

Reply:
I don't understand what info you need, could you please give me a example?